klaas/scripts/apply_separation_model.py

```diff
@@ -1,74 +1,115 @@
-import numpy as np
 import click
-# from IPython import embed
-import joblib
-from klaas import check_extension, read_data, write_data
+from sklearn.externals import joblib
 import yaml
+import logging
+import h5py
+from tqdm import tqdm
+
+from ..io import read_h5py_chunked
+from ..features import find_used_source_features
+from ..apply import predict, predict_off_positions
+
 
 @click.command()
-@click.argument('configuration_path', type=click.Path(exists=True, dir_okay=False, file_okay=True) )
-@click.argument('data_path', type=click.Path(exists=True, dir_okay=False, file_okay=True) )
-@click.argument('model_path', type=click.Path(exists=True, dir_okay=False, file_okay=True) )
-@click.argument('output_path', type=click.Path(exists=False, dir_okay=False, file_okay=True) )
-def main(configuration_path, data_path, model_path, output_path):
+@click.argument('configuration_path', type=click.Path(exists=True, dir_okay=False))
+@click.argument('data_path', type=click.Path(exists=True, dir_okay=False))
+@click.argument('model_path', type=click.Path(exists=True, dir_okay=False))
+@click.option('-k', '--key', help='HDF5 key for pandas or h5py hdf5', default='events')
+@click.option(
+    '-N', '--chunksize', type=int,
+    help='If given, only process the given number of events at once'
+)
+@click.option('-y', '--yes', help='Do not prompt for overwrites', is_flag=True)
+def main(configuration_path, data_path, model_path, key, chunksize, yes):
     '''
-    Apply loaded model to data. The cuts applied during model training will also be applied here.
+    Apply loaded model to data.
 
     CONFIGURATION_PATH: Path to the config yaml file.
-
     DATA_PATH: path to the FACT data.
-
     MODEL_PATH: Path to the pickled model.
 
-    OUTPUT_PATH: Path to the data with added prediction columns.
+    The program adds the following columns to the inputfile:
+        signal_prediction: the output of model.predict_proba for the signal class
     '''
-    check_extension(output_path)
+    logging.basicConfig(level=logging.INFO)
+    log = logging.getLogger()
 
     with open(configuration_path) as f:
         config = yaml.load(f)
 
     training_variables = config['training_variables']
-    query = config['query']
 
+    with h5py.File(data_path) as f:
```

Comment:
shouldn't this also be able to open pandas_hdf5 files? if I try to apply a trained model to a pandas_hdf5 it can naturally not find certain keys